x-pack/plugins/observability_solution/investigate_app/public/components/add_from_library_button/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { createEmbeddableWidget } from '../../widgets/embeddable_widget/create_embeddable_widget';
+import { InvestigateTextButton } from '../investigate_text_button';
+
+interface AddFromLibraryButtonProps {
+  onWidgetAdd: (widget: InvestigateWidgetCreate) => Promise<void>;
+}
+
+export function AddFromLibraryButton({ onWidgetAdd }: AddFromLibraryButtonProps) {
+  const children$ = useMemo(() => new BehaviorSubject({}), []);
+
+  const {
+    dependencies: {
+      start: { contentManagement },
+    },
+  } = useKibana();
+
+  const onWidgetAddRef = useRef(onWidgetAdd);
+
+  onWidgetAddRef.current = onWidgetAdd;
+
+  const panelRef = useRef<OverlayRef>();
+
+  const container = useMemo<
+    PresentationContainer & {
+      addNewEmbeddable: (
+        type: string,
+        explicitInput: { savedObjectId: string },
+        attributes: FinderAttributes
+      ) => Promise<{ id: undefined }>;
+    }
+  >(() => {
+    function addEmbeddable({
+      type,
+      title,
+      description,
+      attributes,
+      savedObjectId,
+    }: {
+      type: string;
+      title: string;
+      description?: string;
+      attributes: Record<string, any>;
+      savedObjectId: string;
+    }) {
+      const widget = createEmbeddableWidget({
+        title,
+        description,
+        parameters: {
+          savedObjectId,
+          config: attributes,
+          type,
+        },
+      });
+
+      onWidgetAddRef.current(widget).then(() => {
+        if (panelRef.current) {
+          panelRef.current.close();
+        }
+      });
+    }
+    return {
+      addNewPanel: async (panel, displaySuccessMessage) => {
+        const state = panel.initialState! as {
+          savedObjectId: string;
+        };
+
+        const savedObject = (await contentManagement.client.get({
+          contentTypeId: panel.panelType,
+          id: state.savedObjectId,
+        })) as { item: { attributes: { title: string } } };
+
+        addEmbeddable({
+          type: panel.panelType,
+          savedObjectId: state.savedObjectId,
+          attributes: {},
+          title: savedObject.item.attributes.title,
+          description: '',
+        });
+
+        return undefined;
+      },
+      removePanel: async (...args) => {
+        throw new Error('removePanel not supported in this context');
+      },
+      replacePanel: async (...args) => {
+        throw new Error('replacePanel not supported in this context');
+      },
+      getPanelCount() {
+        return 0;
+      },
+      addNewEmbeddable: async (type, explicitInput, attributes) => {
+        addEmbeddable({
+          type,
+          title: attributes.title ?? '',
+          description:
+            'description' in attributes && typeof attributes.description === 'string'
+              ? attributes.description
+              : '',
+          savedObjectId: explicitInput.savedObjectId,
+          attributes,
+        });
+        return undefined as any; // { id: undefined };
```

Comment:
I did this to hide the toast from the embeddable API, but it causes an error - we should talk to the Presentation team about this